widgets/common/src/inputs/fields/NumberInputField.tsx

```diff
@@ -0,0 +1,39 @@
+import RevertToDefaultIcon from './RevertToDefaultIcon';
+import { ErrorAwareInputFieldProps, RevertableInputFieldProps } from './types';
+
+type NumberInputField = ErrorAwareInputFieldProps &
+    RevertableInputFieldProps & {
+        type: 'integer' | 'float';
+        constraints: {
+            inRange: [number, number];
+            greaterThan: number;
+            greaterOrEqual: number;
+            lessThan: number;
+            lessOrEqual: number;
+        };
+    };
+
+export default function NumberInputField(props: NumberInputField) {
+    const { Form } = Stage.Basic;
+    const { name, value, onChange, type, error, constraints } = props;
```

Comment:
We can destructurize props as it has been done with the other, extracted input fields 😉 

```suggestion
export default function NumberInputField({
    name,
    value,
    onChange,
    type,
    error,
    constraints
}: NumberInputField) {
    const { Form } = Stage.Basic;
```

Reply:
See L35 - `props` object is used there directly.

Reply:
Oh yea, you're right!
Thanks for the clarification :)